tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -0,0 +1,317 @@
+            .map(str -> str.replaceAll("'", "").trim()).collect(Collectors.toSet());
+
+        assertTrue(output.matches("Deletion of requested consumer groups (.*) was successful.")
+            && Objects.equals(deletedGroupsGrepped, expectedGroupsForDeletion),
+            "The consumer group(s) could not be deleted as expected");
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testDeleteEmptyGroup(String quorum) throws Exception {
+        createOffsetsTopic(listenerName(), new Properties());
+
+        // run one consumer in the group
+        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1);
+        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP};
+        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+
+        TestUtils.waitForCondition(
+            () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state(), "Stable"),
+            "The group did not initialize as expected.");
+
+        executor.shutdown();
+
+        TestUtils.waitForCondition(
+            () -> Objects.equals(service.collectGroupState(GROUP).state(), "Empty"),
+            "The group did not become empty as expected.");
+
+        scala.collection.Map<String, Throwable> result = service.deleteGroups();
+        assertTrue(result.size() == 1 && result.contains(GROUP) && result.get(GROUP).get() == null,
+            "The consumer group could not be deleted as expected");
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testDeleteCmdWithMixOfSuccessAndError(String quorum) throws Exception {
+        createOffsetsTopic(listenerName(), new Properties());
+        String missingGroup = "missing.group";
+
+        // run one consumer in the group
+        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1);
+        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP};
+        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+
+        TestUtils.waitForCondition(
+            () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state(), "Stable"),
+            "The group did not initialize as expected.");
+
+        executor.shutdown();
+
+        TestUtils.waitForCondition(
+            () -> Objects.equals(service.collectGroupState(GROUP).state(), "Empty"),
+            "The group did not become empty as expected.");
+
+        String[] cgcArgs2 = new String[cgcArgs.length + 2];
+
+        System.arraycopy(cgcArgs, 0, cgcArgs2, 0, cgcArgs.length);
+        cgcArgs2[cgcArgs2.length - 2] = "--group";
+        cgcArgs2[cgcArgs2.length - 1] = missingGroup;
```

Comment:
I know we did that in scala like this, but after rewriting in Java it becomes more unreadable. Maybe we can just do this:
```
String[] cgcArgs2 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP, "--group", missingGroup};
```

Reply:
Done.